core/modules/taxonomy/taxonomy.admin.inc

```diff
@@ -17,64 +17,74 @@ function taxonomy_overview_vocabularies($form) {
+          '#title_display' => 'invisible',
+          '#delta' => 10,
+          '#default_value' => $vocabulary->weight,
+        );
+      }
+      $links = array();
+      $links['list'] = array(
+        'title' => t('List terms'),
+        'href' => "admin/structure/taxonomy/$vocabulary->machine_name",
       );
-      $links['display'] = array(
-        'title' => t('Manage displays'),
-        'href' => "admin/structure/taxonomy/$vocabulary->machine_name/display",
-        'weight' => 10,
+      if (taxonomy_vocabulary_access('create', $vocabulary)) {
+        $links['add'] = array(
+          'title' => t('Add terms'),
+          'href' => "admin/structure/taxonomy/$vocabulary->machine_name/add",
+        );
+      }
+      if (user_access('administer taxonomy')) {
+        $links['configure'] = array(
+          'title' => t('Configure'),
+          'href' => "admin/structure/taxonomy/$vocabulary->machine_name/configure",
+        );
+      }
+      if (module_exists('field_ui') && user_access('administer fields')) {
+        $links['fields'] = array(
+          'title' => t('Manage fields'),
+          'href' => "admin/structure/taxonomy/$vocabulary->machine_name/fields",
+          'weight' => 5,
+        );
+        $links['display'] = array(
+          'title' => t('Manage displays'),
+          'href' => "admin/structure/taxonomy/$vocabulary->machine_name/display",
+          'weight' => 10,
+        );
+      }
+      if (user_access('synchronize configuration')) {
+        $links['export'] = array(
+          'title' => t('Export'),
+          'href' => 'admin/config/development/configuration/single/export',
+          'query' => array(
+            'group' => 'Vocabularies',
+            'name' => 'taxonomy.vocabulary.' . $vocabulary->machine_name,
+          ),
+        );
+      }
+      if (user_access('administer taxonomy')) {
+        $links['delete'] = array(
+          'title' => t('Delete'),
+          'href' => "admin/structure/taxonomy/$vocabulary->machine_name/delete",
         );
       }
-    if (user_access('synchronize configuration')) {
-      $links['export'] = array(
-        'title' => t('Export'),
-        'href' => 'admin/config/development/configuration/single/export',
-        'query' => array(
-          'group' => 'Vocabularies',
-          'name' => 'taxonomy.vocabulary.' . $vocabulary->machine_name,
-        ),
+      $form[$vocabulary->machine_name]['operations'] = array(
+        '#type' => 'operations',
+        '#links' => $links,
       );
     }
-    $links['delete'] = array(
-      'title' => t('Delete'),
-      'href' => "admin/structure/taxonomy/$vocabulary->machine_name/delete",
-    );
-    $form[$vocabulary->machine_name]['operations'] = array(
-      '#type' => 'operations',
-      '#links' => $links,
-    );
   }
 
   // Only make this form include a submit button and weight if more than one
   // vocabulary exists.
-  if (count($vocabularies) > 1) {
```

Comment:
This might conflict since we've removed the form from the vocabulary overview form.